### PR TITLE
Issue#13 User model

### PR DESCRIPTION
**Pull request by radowit**

### CODE REVIEW

File: app/crowd/tests.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        self.assertIsNone(user1.clean())
+
+    def test_username_is_not_unique(self):
+        user1 = User.objects.create(username="", first_name="Maciek")
+        user2 = User.objects.create(username="", first_name="Maciek")
+
+        self.assertEqual(user1.username, user2.username)
+
+    def test_username_or_first_name_should_be_provided(self):
+        self.assertRaisesRegex(
```
Comment (lprasol) [inline app/crowd/tests.py]: IMO using `assertRaisesRegex` context manager looks more readable.

File: app/crowd/tests.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        )
+
+    def test_username_is_not_required_in_manager(self):
+        user = User.objects.create_user(first_name="Maciek")
+
+        self.assertEqual(user.first_name, "Maciek")
+        self.assertEqual(user.username, "")
+
+    def test_username_or_first_name_should_be_provided_in_manager(self):
+        self.assertRaisesRegex(
```
Comment (lprasol) [inline app/crowd/tests.py]: Suggestion: context manager?

File: .pylintrc [hunk 3]
```diff
@@ -62,7 +62,7 @@ confidence=
 # --enable=similarities". If you want to run only the classes checker, but have
 # no Warning level messages displayed, use "--disable=all --enable=classes
 # --disable=W".
-disable=missing-docstring
+disable=missing-docstring,bad-continuation
```
Comment (lprasol) [inline .pylintrc]: Why? :(

File: app/crowd/models.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        email: Optional[str] = None,
+        password: Optional[str] = None,
+        **extra_fields: Any,
+    ) -> AbstractUser:
+        """Create user requiring username or first_name."""
+        return super().create_user(  # type: ignore
+            username, email, password, **extra_fields
+        )
+
+    def _create_user(
```
Comment (lprasol) [inline app/crowd/models.py]: Why do we need two methods? Private and a public? Probably I missed something...

Comment (radowit) [reply app/crowd/models.py]: These are methods from original `UserManager` class.

File: .pylintrc (hunk 3 above)
Comment (radowit) [reply .pylintrc]: Because black has different continuation rules than pylint.

File: app/crowd/tests.py (hunk 1 above)
Comment (radowit) [reply app/crowd/tests.py]: Good point. I love context managers.

File: app/crowd/migrations/0002_auto_20200225_1947.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# Generated by Django 2.2.10 on 2020-02-25 18:47
```
[the remark this answers is not in the document]
Comment (radowit) [reply app/crowd/migrations/0002_auto_20200225_1947.py]: oks

File: app/crowd/models.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    USERNAME_FIELD = "uuid"
+
+    objects = CrowdUserManager()
+
+    class Meta:  # noqa: D106
+        verbose_name = _("user")
+        verbose_name_plural = _("users")
+        db_table = "cr_user"
+
+    def __str__(self) -> str:
```
[the remark this answers is not in the document]
Comment (radowit) [reply app/crowd/models.py]: Ok, I'll return the result of `get_full_name` and change this function to display name in format:
`John Kowalsky (darklord85)`.

File: app/crowd/models.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    def _create_user(
+        self,
+        username: str = "",
+        email: Optional[str] = None,
+        password: Optional[str] = None,
+        **extra_fields: Any,
+    ) -> AbstractUser:
+        """Create and save a user with the given username, email, and password."""
+        if not username and not extra_fields.get("first_name"):
+            raise ValidationError("Please specify either username or first_name.")
```
[the remark this answers is not in the document]
Comment (radowit) [reply app/crowd/models.py]: sure

**Pull request merged**